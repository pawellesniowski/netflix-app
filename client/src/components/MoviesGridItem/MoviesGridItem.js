import React from 'react';
import './MoviesGridItem.scss';
import MovieTitle from "../MovieTitle/MovieTitle";
import MovieReleaseDate from "../MovieReleaseDate/MovieReleaseDate";
import MovieGenres from "../MovieGenres/MovieGenres";

const MoviesGridItem = (props) => {
  const {poster_path, title, release_date, genres, id} = props.movie;
  return (
    <li className="movies-grid-item">
      <img className="movies-grid-item-image" src={poster_path}/>
      <div className="movies-grid-item-header">
        <MovieTitle title={title} />
        <MovieReleaseDate release_date={release_date} />
      </div>
      <MovieGenres genres={genres}/>
    </li>
  );
};

export default MoviesGridItem;
