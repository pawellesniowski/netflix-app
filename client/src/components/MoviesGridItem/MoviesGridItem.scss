.movies-grid-item {
  font-size: 1.6rem;
  list-style: none;
  padding: 1%;
  width: 30%;

  .movies-grid-item-image {
    width: 100%;
  }

  .movies-grid-item-header {
    display: flex;
    justify-content: space-between;
  }
}

