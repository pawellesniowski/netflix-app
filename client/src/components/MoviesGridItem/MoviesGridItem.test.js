import React from 'react';
import {shallow} from 'enzyme';
import MoviesGridItem from './MoviesGridItem';

describe("MoviesGridItem Component", () => {
    let component;
    beforeEach(() => {
        component = shallow(<MoviesGridItem/>);
    });

    it("Renders the container", () => {
        expect(component.find('[data-cy="movies-grid-item"]')).toHaveLength(1);
    });

    it("Renders all the elements", () => {
        expect(component.find('.movies-grid-item-image')).toHaveLength(1);
        expect(component.find('.movies-grid-item-header')).toHaveLength(1);
        expect(component.find('.movies-grid-item-header')).toHaveLength(1);
        expect(component.find('MovieTitle')).toHaveLength(1);
        expect(component.find('MovieReleaseDate')).toHaveLength(1);
        expect(component.find('MovieGenres')).toHaveLength(1);
    });

    it("Matches the snapshot", () => {
        expect(component).toMatchSnapshot();
    });
});
