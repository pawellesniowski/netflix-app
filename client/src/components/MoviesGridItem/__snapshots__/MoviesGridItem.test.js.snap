// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesGridItem Component Matches the snapshot 1`] = `ShallowWrapper {}`;
